nixos/modules/services/networking/pixiecore.nix

```diff
@@ -0,0 +1,114 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.pixiecore;
+
+in
+
+{
+
+  options = {
+
+    services.pixiecore = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to run pixiecore netboot server.";
+      };
+
+     debug = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Log more things that aren't directly related to booting a recognized client";
+      };
+
+     logTimestamps = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Add a timestamp to each log line";
+      };
+
+     dhcpNoBind = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Handle DHCP traffic without binding to the DHCP server port";
+      };
+
+      bootMsg = mkOption {
+        type = types.nullOr types.string;
+        default = "";
+        description = "Message to print on machines before booting";
+      };
+
+      kernel = mkOption {
+        type = types.string;
+        default = "";
+        description = "Kernel path";
+      };
+
+      initrd = mkOption {
+        type = types.string;
+        default = "";
+        description = "Initrd path";
+      };
+
+      cmdLine = mkOption {
+        type = types.nullOr types.string;
+        default = "";
+        description = "Kernel commandline arguments";
+      };
+
+     listen = mkOption {
+        type = types.string;
+        default = "0.0.0.0";
+        description = "IPv4 address to listen on";
+      };
+
+     port = mkOption {
+        type = types.int;
+        default = 80;
+        description = "Port to listen on for HTTP";
+      };
+
+     statusPort = mkOption {
+        type = types.int;
+        default = 80;
+        description = "HTTP port for status information (can be the same as --port)";
+      };
+
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+#    users.users.pixiecore = {
```

Comment:
Am I correct in that you were using a user for the systemd service but then switched to root because of port < 1024 issues? Fortunately modern distros like NixOS allow you to run services which can bind to ports < 1024 without requiring root so I suggest you look into that as running services as root is highly discouraged if not necessary.